android-sdk/src/androidTest/java/com/mobilecoin/lib/TxOutMemoIntegrationTest.java

```diff
@@ -20,446 +23,202 @@
-        .parseTxOutMemo(changeMemoPayload, senderAccountKey, changeTxOut);
-    DestinationMemoData destinationMemoData = destinationMemo.getDestinationMemoData();
-    UnsignedLong totalOutlay = UnsignedLong.valueOf(fee.add(txAmount).getValue());
-
-    assertEquals(UnsignedLong.valueOf(fee.getValue()), destinationMemoData.getFee());
-    assertEquals(totalOutlay, destinationMemoData.getTotalOutlay());
-    AddressHash recipientAddressHash = recipientAccountKey.getPublicAddress()
-        .calculateAddressHash();
-    assertEquals(recipientAddressHash, destinationMemoData.getAddressHash());
-  }
-
-  @Test
-  public void buildTransaction_senderWithPaymentRequestAndDestinationMemoBuilder_buildsCorrectSenderWithPaymentRequestMemo() throws Exception {
-    UnsignedLong paymentRequestId = UnsignedLong.valueOf(301);
-    TxOutMemoBuilder txOutMemoBuilder = TxOutMemoBuilder
-        .createSenderPaymentRequestAndDestinationRTHMemoBuilder(senderAccountKey, paymentRequestId);
-    TxOut realTxOut = txOuts.get(realIndex);
-    Amount fee = new Amount(BigInteger.ONE, TokenId.MOB);
-    transactionBuilder = new TransactionBuilder(
-            fogResolver,
-            txOutMemoBuilder,
-            MEMO_BLOCK_VERSION,
-            TokenId.MOB,
-            fee
-          );
-
-    RistrettoPrivate onetimePrivateKey = OnetimeKeys.recoverOnetimePrivateKey(
-        realTxOut.getPublicKey(),
-        realTxOut.getTargetKey(),
-        senderAccountKey
-    );
-    transactionBuilder
-        .addInput(txOuts, txOutMembershipProofs, realIndex, onetimePrivateKey,
-            senderAccountKey.getViewKey());
-
-    transactionBuilder.setTombstoneBlockIndex(UnsignedLong.valueOf(2000));
-    Amount sentTxOutAmount = Amount.ofMOB(BigInteger.ONE);
-
-    transactionBuilder.addOutput(sentTxOutAmount, recipientAccountKey.getPublicAddress(), null);
-    Amount realTxOutAmount = realTxOut.getMaskedAmount()
-        .unmaskAmount(senderAccountKey.getViewKey(), realTxOut.getPublicKey());
-    Amount change = realTxOutAmount.subtract(fee)
-        .subtract(sentTxOutAmount);
-    transactionBuilder.addChangeOutput(change, senderAccountKey, null);
-    Transaction transaction = transactionBuilder.build();
-
-    List<MobileCoinAPI.TxOut> outputsList = transaction.toProtoBufObject().getPrefix()
-        .getOutputsList();
-    TxOut txOut1 = TxOut.fromProtoBufObject(outputsList.get(0));
-    TxOut txOut2 = TxOut.fromProtoBufObject(outputsList.get(1));
-
-    TxOut sentTxOut;
-    TxOut changeTxOut;
-    try {
-      txOut1.getMaskedAmount().unmaskAmount(recipientAccountKey.getViewKey(), txOut1.getPublicKey());
-      sentTxOut = txOut1;
-      changeTxOut = txOut2;
-    } catch(Exception e) {
-      sentTxOut = txOut2;
-      changeTxOut = txOut1;
+    private static final int MEMO_BLOCK_VERSION = 1;
+
+    @Test
+    public void txOutMemoIntegrationTest() throws Exception {
+        final MobileCoinClient senderClient = MobileCoinClientBuilder.newBuilder().build();
```

Comment:
A single client is built and refreshed only once. After this, the transactions may be built to test memos without much additional overhead. So test performance isn't severely impacted.